apps/test/unit/templates/manageStudents/MoveStudentsTest.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {expect} from '../../../util/configuredChai';
+import MoveStudents from '@cdo/apps/templates/manageStudents/MoveStudents';
+
+const studentData = [
+  {id: 1, name: 'studentb'},
+  {id: 3, name: 'studenta'},
+  {id: 0, name: ''},
+  {id: 2, name: 'studentf'}
+];
+
+describe('MoveStudents', () => {
+  let wrapper;
+
+  beforeEach(() => {
+    wrapper = shallow(<MoveStudents studentData={studentData}/>);
+  });
+
+  describe('#openDialog', () => {
+    it('sets isDialogOpen state to true', () => {
+      wrapper.instance().openDialog();
+      expect(wrapper.instance().state.isDialogOpen).to.equal(true);
+    });
+  });
+
+  describe('#closeDialog', () => {
+    it('sets isDialogOpen state to false', () => {
+      wrapper.instance().closeDialog();
+      expect(wrapper.instance().state.isDialogOpen).to.equal(false);
```

Comment:
:laughing: Here you've tested that the dialog _stays_ closed when it's already closed and you call `closeDialog()`.  That is an important behavior, but maybe not the one you meant to cover?

This is also interesting because in your implementation those two cases are the same code path - but I could imagine a world where some future developer refactors this component and breaks one case but not the other.

Reply:
😱 copy-pasting tests wins again